vmware-horizon-client: 2406 -> 2503, rename to omnissa-horizon-client

mhutter opened the pull request:
This PR renames `vmware-horizon-client` to `omnissa-horizon-client` and updates it to the latest Version.

VMware Horizon was sold to Omnissa in 2024, and has since been rebranded as Omnissa Horizon.

Source: https://www.omnissa.com/insights/introducing-omnissa-the-former-vmware-end-user-computing-business/

Release Notes: https://docs.omnissa.com/bundle/horizon-client-linux-rnV2503/page/HorizonClientLinux-ReleaseNotes.html

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] ~aarch64-linux~
  - [ ] ~x86_64-darwin~
  - [ ] ~aarch64-darwin~
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] ~`sandbox = relaxed`~
  - [ ] ~`sandbox = true`~
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mhutter committed:
vmware-horizon-client -> omnissa-horizon-client

VMware Horizon was sold to Omnissa in 2024, and has since been rebranded
as Omnissa Horizon.

Source: https://www.omnissa.com/insights/introducing-omnissa-the-former-vmware-end-user-computing-business/

Signed-off-by: Manuel Hutter <manuel@hutter.io>

mhutter commented:
Been running this for a week without issues :+1:

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5434

undefined-landmark commented:
Just fyi, maybe you're already aware, similar work has been done in the following PR, but it seems that work has been stalled: https://github.com/NixOS/nixpkgs/pull/377554. Personally I use this package almost daily so hopefully I can be of some help to move your PR forward. Some points:
- Release notes are missing
- The `top-level/aliases.nix` warning seems pretty neat. Nicer than keeping around the vmware-horizon-client files, just for throwing a warning, like in the other commit.

Other than that the changes seem pretty minimal. Looks good to me.

06kellyjac commented on pkgs/by-name/om/omnissa-horizon-client/package.nix:
```diff
@@ -145,19 +145,19 @@ stdenv.mkDerivation {
+    homepage = "https://www.omnissa.com/products/horizon-8/";
     license = licenses.unfree;
     platforms = [ "x86_64-linux" ];
     maintainers = [ ];
```
Are you willing to be a maintainer?

mhutter replied:
Yes for the time being I'll have to work with this anyway.

06kellyjac reviewed (approved):
Change looks good.
I like the throw in aliases & fresh package.

Added question about maintenance.

mhutter commented:
@06kellyjac can I just add myself as a maintainer?

mhutter committed:
maintainers: add mhutter

Signed-off-by: Manuel Hutter <manuel@hutter.io>

06kellyjac reviewed (changes requested):
Looks good, just needs a release note for 25.11

mhutter committed:
omnissa-horizon-client: fix libxml2 build incompatibility

Breaking ABI change for libxml2 recently, need to override to use
supported libxml version.

Signed-off-by: Manuel Hutter <manuel@hutter.io>

mhutter commented:
@06kellyjac Changelog done,

@undefined-landmark yes I just noticed this morning that I forgot to incorporate that fix here.

undefined-landmark commented:
Was connected to a VM via this application for about 8 hours today. No issues. 

Small sidenote: I only use the basics, no virtual USB drives or transferring data between the host machine and VM etc..

06kellyjac reviewed (approved):
I've also tested this PR with the basics and was working fine.
The only issue I had was my desktop within the client wasn't resizing for a bit when I first connected but it resolved itself & wouldn't be an issue with the packaging.

LGTM

---

One thing for future would be to follow this guidance when adding to the release notes

> To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead.

undefined-landmark commented:
> The only issue I had was my desktop within the client wasn't resizing for a bit when I first connected but it resolved itself & wouldn't be an issue with the packaging.

I did notice that as well. But a second resize fixed it. Agree that it's unlikely an issue with the packaging.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5649

undefined-landmark commented:
There is a merge conflict atm. Probably due to the release notes?

corngood commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -1,6 +1,7 @@
 # Nixpkgs 25.11 ("Xantusia", 2025.11/??) {#sec-nixpkgs-release-25.11}
 
 ## Highlights {#sec-nixpkgs-release-25.11-highlights}
+
```
unintended whitespace change?

mhutter replied:
Yes, from my autoformatter; removed from commit

corngood commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -70,7 +73,6 @@
 
 - Create the first release note entry in this section!
 
-
```
unintended whitespace change?

mhutter replied:
Same as above

corngood commented on pkgs/by-name/om/omnissa-horizon-client/package.nix:
```diff
@@ -116,6 +115,22 @@ let
           xorg.libXtst
           zlib
 
+          # c.f. https://github.com/NixOS/nixpkgs/pull/418543
```
It's a shame we have to maintain this here, but at least there's precedent, and there doesn't seem to be an appetite to do it globally (https://github.com/NixOS/nixpkgs/pull/396195).

Hopefully this won't stick around for too long.

mhutter replied:
Yes I'm not too fond of this either... My hope is that Omnissa will release a new version that is compatible with the latest libxml2 _eventually_ and we can get rid of that ...

corngood reviewed (commented):
I've tested this and it works at least as well as the old version for me.

I'd be happy to merge this, but could you rebase on master to remove the merge commit?

corngood commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401532`
Commit: `991e3576287876e15ec9f7b206810a0f5213598f`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>nixpkgs-manual</li>
    <li>omnissa-horizon-client</li>
  </ul>
</details>

corngood reviewed (approved):
Looks good.  Thanks!
